iceoryx_hoofs/source/posix_wrapper/timer.cpp

```diff
@@ -166,11 +161,8 @@ void Timer::OsTimer::callbackHelper(sigval data) noexcept
     }
     else
     {
-        if (handle.m_catchUpPolicy == CatchUpPolicy::TERMINATE)
-        {
-            errorHandler(Error::kPOSIX_TIMER__CALLBACK_RUNTIME_EXCEEDS_RETRIGGER_TIME);
-        }
-        return;
+        cxx::Ensures((handle.m_catchUpPolicy != CatchUpPolicy::TERMINATE)
```

Comment:
Personally I think the distinction between `Ensures`and `Expects` is arbitrary and not needed. Especially if I can and want to call either in the middle of a function. The names are also not meaningfully distinct.

We just need one of them and might as well call it `Assert` (though I am fine with `Expect` as well). Just a better flow when reading the code IMO (also closer to say GTest, where you also *expect* some condition to hold at specific points of the test).

Reply:
I put `Requires` to the polls :laughing: